core/src/main/scala/higherkindness/mu/rpc/srcgen/GeneratorApplication.scala

```diff
@@ -37,16 +41,36 @@ class GeneratorApplication[T <: Generator](generators: T*) {
       inputFiles: Set[File],
       outputDir: File
   ): Seq[File] =
-    if (idlTypes.contains(idlType))
-      generatorsByType(idlType).generateFrom(inputFiles, serializationType).map {
-        case (inputFile, outputFilePath, output) =>
-          val outputFile = new File(outputDir, outputFilePath)
-          logger.info(s"$inputFile -> $outputFile")
-          Option(outputFile.getParentFile).foreach(_.mkdirs())
-          outputFile.write(output)
-          outputFile
+    if (idlTypes.contains(idlType)) {
+      val result: ValidatedNel[(File, NonEmptyList[Error]), List[File]] =
+        generatorsByType(idlType)
+          .generateFrom(inputFiles, serializationType)
+          .traverse {
+            case (inputFile, outputFilePath, output) =>
+              output match {
+                case Invalid(readErrors) =>
+                  (inputFile, readErrors).invalidNel
+                case Valid(content) =>
+                  val outputFile = new File(outputDir, outputFilePath)
+                  logger.info(s"$inputFile -> $outputFile")
+                  Option(outputFile.getParentFile).foreach(_.mkdirs())
+                  outputFile.write(content)
+                  outputFile.validNel
+              }
+          }
+      result match {
+        case Invalid(listOfFilesAndReadErrors) =>
+          val formattedErrorMessage = listOfFilesAndReadErrors.map {
+            case (inputFile, errors) =>
+              s"$inputFile has the following errors: ${errors.toList.mkString(", ")}"
+          }
```

Comment:
How about one line per file? This also means we don't include `NonEmptyList(...)` in our output:

```suggestion
          }.mkString("\n")
```